### ISSUE TITLE
Fedora 24: fstl does not run with qt5 and does not compile with qt4

### ISSUE THREAD

**Issue by graywolf2** [mkeeter/fstl#8]
My operating system is Fedora 24 64 bit. Videocard is Nvidia NVS 5200M. I use the proprietary nvidia driver.
Qt version is 5.6.1.

Qt5:
I compiled the fstl normally by qmake-qt5 and make commands. But fstl does not run.
vovk@dell:/mnt/datadisk1/Work/fstl-master/release$ ./fstl
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
----------------------------------------------------------

With qt4 the fstl does not compile. I attached the log.

[qt4_compile_log.txt](https://github.com/mkeeter/fstl/files/459778/qt4_compile_log.txt)

**Comment by graywolf2** [comment]
If in app.cpp I do this:
`App::App(int argc, char *argv[]) :
	QApplication(argc, argv), window(new Window())
{
    window->show();
	/*
    if (argc > 1)
        window->load_stl(argv[1]);
    else
        window->load_stl(":gl/sphere.stl");
	*/
}`
the fstl runs normally but no model is shown in window.

So bug is in Window::load_stl function or in functions that execute Window::load_stl.

**Comment by graywolf2** [comment]
I had fixed the primary bug!

The QApplication constructor is defined as
QApplication(int &argc, char **argv)
So we must to pass to this constructor the same argc instance as from main function, not the copy.

I had changed the App::App(int argc, char *argv[]) definition to
App::App(int &argc, char *argv[])

And now fstl runs normally. But I see new bug. There is no 3d model in window, only black area. I attached the screenshot.
![screenshot](https://cloud.githubusercontent.com/assets/12239412/18327156/db2e4232-7551-11e6-985c-c89d3ba3486b.png)

**Comment by graywolf2** [comment]
From documentation about QApplication constructor:
Warning: The data referred to by argc and argv must stay valid for the entire lifetime of the QApplication object.

Before my fix the argc variable in App constructor did not live long. :)

**Comment by graywolf2** [comment]
The "black area" bug is related to nvidia proprietary driver.
BTW the 3d game like flightgear works normally with this driver.

I loaded the live Fedora 24 with opensource nouveau driver from flash drive. The fstl worked normally. I attached screenshots.
![screenshot1](https://cloud.githubusercontent.com/assets/12239412/18330475/82cc4746-7561-11e6-85c0-5d00fcdece27.png)
![screenshot2](https://cloud.githubusercontent.com/assets/12239412/18330485/91a00f96-7561-11e6-8c7b-f9591fc09d55.png)
